resources/definitions/KOONOVOELF.def.json

```diff
@@ -0,0 +1,94 @@
+{
+    "version": 2,
+    "name": "KOONOVO ELF",
+    "inherits": "fdmprinter",
+    "metadata": {
+        "visible": true,
+        "author": "KOONOVO",
+        "manufacturer": "KOONOVO",
+        "weight": 4,
+        "file_formats": "text/x-gcode",
+        "icon": "icon_ultimaker.png",
+
+        "platform": "KOONOVO.obj",
+        "platform_texture": "KOONOVO.png",
+        "has_materials": true,
+        "preferred_material": "generic_pla",
+        "has_machine_quality": true,
+        "quality_definition": "KOONOVO_ELF",
```

Comment:
KOONOVO_ELF is your extruder definition. You cannot refer to the extruder definition for qualities.